library/std/src/sys/pal/sgx/abi/usercalls/alloc.rs

```diff
@@ -266,9 +267,8 @@ where
     /// Converts this value into a raw pointer. The value will no longer be
     /// automatically freed.
     pub fn into_raw(self) -> *mut T {
-        let ret = self.0;
-        mem::forget(self);
-        ret.as_ptr() as _
+        let Self(nonnull) = self;
```

Comment:
The inner `NonNull` is `Copy` so I think `self` still gets dropped after this line.

Reply:
Oh, you are right! I have totally forgotten how destructuring works.